File: 1460-number-of-substrings-containing-all-three-characters/1460-number-of-substrings-containing-all-three-characters.cpp
class Solution {
public:
    int numberOfSubstrings(string s) {
        int n = s.size(), ans = 0 ;
        int i = 0 , j = 0 ; 
        map<char,int>mp;
        while(j<n){
            while(mp.size() < 3 && j <n){
                mp[s[j]]++;      
                j++;
            }
            while(mp.size() == 3 && i<j){
                mp[s[i]]--;
                if(mp[s[i]] == 0)mp.erase(s[i]);
                i++;
                ans += n-j+1;
            }
        }
        return ans;

    }
};

File: 1460-number-of-substrings-containing-all-three-characters/1460-number-of-substrings-containing-all-three-characters.cs
public class Solution {
    public bool size(ref List<int>v){
        return v[0] >= 1 && v[1] >= 1 && v[2] >= 1;
    }
    public int NumberOfSubstrings(string s) {
        int n = s.Length;
        int ans = 0 ;
        List<int> v = new List<int>{0,0,0};
        int i = 0 , j = 0;
        while(j<n){
            while(!size(ref v) && j < n ){
                v[s[j]-97]++;
                j++;
            }
            while(size(ref v) && i < j){
                ans += (n - j + 1);
                v[s[i]-97]--;
                i++;
            }
        } 
        return ans;
    }
}